dagger-core/src/test/java/io/odpf/dagger/core/processors/external/es/EsSourceConfigTest.java

```diff
@@ -198,29 +229,53 @@ public void shouldReturnMandatoryFields() {
         expectedMandatoryFields.put("socket_timeout", socketTimeout);
         expectedMandatoryFields.put("stream_timeout", streamTimeout);
         expectedMandatoryFields.put("fail_on_errors", failOnErrors);
-        expectedMandatoryFields.put("outputMapping", defaultOutputMapping);
-        HashMap<String, Object> actualMandatoryFields = defaultEsSourceConfig.getMandatoryFields();
-        assertArrayEquals(expectedMandatoryFields.values().toArray(), actualMandatoryFields.values().toArray());
+        expectedMandatoryFields.put("outputMapping", outputMappingHashMap);
+        EsSourceConfig esSourceConfig = getValidEsSourceConfigBuilder().createEsSourceConfig();
+        HashMap<String, Object> actualMandatoryFields = esSourceConfig.getMandatoryFields();
+        assertEquals(expectedMandatoryFields, actualMandatoryFields);
     }
 
     @Test
     public void shouldValidateWhenOutputMappingIsEmpty() {
         expectedException.expect(IllegalArgumentException.class);
         expectedException.expectMessage("Missing required fields: [outputMapping]");
 
-        EsSourceConfig esSourceConfig = new EsSourceConfig(host, port, user, password, endpointPattern, endpointVariables, type, capacity, connectTimeout, retryTimeout, socketTimeout, streamTimeout, false, new HashMap<>(), metricId, retainResponseType);
-
+        EsSourceConfigBuilder validEsSourceConfigBuilder = getValidEsSourceConfigBuilder();
+        EsSourceConfig esSourceConfig = validEsSourceConfigBuilder
+                .setOutputMapping(Collections.emptyMap())
+                .createEsSourceConfig();
         esSourceConfig.validateFields();
     }
 
     @Test
     public void shouldValidateWhenEndpointPatternIsEmpty() {
         expectedException.expect(IllegalArgumentException.class);
```

Comment:
can we change the exception to argument captor?

Reply:
thanks for pointing out, had upgraded the junit version afterwards so missed changing expected rule to asserthrows here , will update it